tests/unit/EntityId/PrefixMappingEntityIdParserTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\EntityId;
+
+use Wikibase\DataModel\Entity\EntityIdParser;
+use Wikibase\DataModel\Entity\EntityIdParsingException;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\EntityId\PrefixMappingEntityIdParser;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterElementTypeException;
+use Wikimedia\Assert\ParameterTypeException;
+
+/**
+ * @covers Wikibase\DataModel\Services\EntityId\PrefixMappingEntityIdParser
+ *
+ * @license GPL-2.0+
+ */
+class PrefixMappingEntityIdParserTest extends \PHPUnit_Framework_TestCase {
+
+	private function getRegularParser( $input, EntityId $returnValue ) {
+		$regularParser = $this->getMock( EntityIdParser::class );
+		$regularParser->expects( $this->once() )
+			->method( 'parse' )
+			->with( $input )
+			->will( $this->returnValue( $returnValue ) );
+		return $regularParser;
+	}
+
+	/**
+	 * @dataProvider validIdSerializationProvider
+	 */
+	public function testGivenNoPrefixMapping_prefixIsAddedToIdSerialization( $prefix, $input, $expected ) {
+		$parser = new PrefixMappingEntityIdParser(
+			$prefix,
+			[],
+			$this->getRegularParser( $prefix . ':' . $input, $expected )
+		);
+		$this->assertEquals( $expected, $parser->parse( $input ) );
+	}
+
+	public function validIdSerializationProvider() {
```

Comment:
I would prefer provideValidIdSerialization(), since function names should be verbs. But we are sadly inconsistent in that regard, so whatever.